test/celeritas/ext/GeantImporter.test.cc

```diff
@@ -186,6 +188,677 @@ TEST_F(FourSteelSlabsEmStandard, em_hadronic)
+        ASSERT_EQ(ImportTableType::dedx, dedx.table_type);
+        EXPECT_EQ(ImportUnits::mev, dedx.x_units);
+        EXPECT_EQ(ImportUnits::mev_per_cm, dedx.y_units);
+        ASSERT_EQ(2, dedx.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = dedx.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(85, steel.x.size());
+        EXPECT_SOFT_EQ(1e-4, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(839.66835335480653, steel.y.front());
+        EXPECT_SOFT_EQ(11.380485677836065, steel.y.back());
+    }
+    {
+        // Test range table
+        const ImportPhysicsTable& range = tables[1];
+        ASSERT_EQ(ImportTableType::range, range.table_type);
+        EXPECT_EQ(ImportUnits::mev, range.x_units);
+        EXPECT_EQ(ImportUnits::cm, range.y_units);
+        ASSERT_EQ(2, range.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = range.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(85, steel.x.size());
+        EXPECT_SOFT_EQ(1e-4, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(2.3818927937550707e-07, steel.y.front());
+        EXPECT_SOFT_EQ(8786971.3079055995, steel.y.back());
+    }
+    {
+        // Test cross-section table
+        const ImportPhysicsTable& xs = tables[2];
+        ASSERT_EQ(ImportTableType::lambda, xs.table_type);
+        EXPECT_EQ(ImportUnits::mev, xs.x_units);
+        EXPECT_EQ(ImportUnits::cm_inv, xs.y_units);
+        ASSERT_EQ(2, xs.physics_vectors.size());
+
+        const ImportPhysicsVector& steel = xs.physics_vectors.back();
+        EXPECT_EQ(ImportPhysicsVectorType::log, steel.vector_type);
+        ASSERT_EQ(steel.x.size(), steel.y.size());
+        ASSERT_EQ(54, steel.x.size());
+        EXPECT_SOFT_EQ(2.6269057995911775, steel.x.front());
+        EXPECT_SOFT_EQ(1e8, steel.x.back());
+        EXPECT_SOFT_EQ(0, steel.y.front());
+        EXPECT_SOFT_EQ(0.18987862452122845, steel.y[1]);
+        EXPECT_SOFT_EQ(0.43566778103861714, steel.y.back());
+    }
+    {
+        // Test model microscopic cross sections
+        auto get_values = [](const ImportProcess::ElementPhysicsVectors& xs) {
+            std::unordered_map<std::string, std::vector<double>> result;
+            for (const auto& vec : xs)
+            {
+                result["x_size"].push_back(vec.x.size());
+                result["y_size"].push_back(vec.y.size());
+                result["x_front"].push_back(vec.x.front());
+                result["y_front"].push_back(vec.y.front() / units::barn);
+                result["x_back"].push_back(vec.x.back());
+                result["y_back"].push_back(vec.y.back() / units::barn);
+            }
+            return result;
+        };
```

Comment:
The comment I made about this might've gotten lost in the shuffle regarding element IDs... but a `struct` here is much more appropriate than a `map` since the keys are known and the value types should be different.

Could you update the test using the helper code:
```c++
#include "corecel/io/Repr.hh"

struct ImportedXsSummary
{
    std::vector<size_type> size;
    std::vector<real_type> x_bounds;
    std::vector<real_type> y_bounds;

    void print_expected() const;
};

ImportedXsSummary summarize(const ImportProcess::ElementPhysicsVectors& xs)
{
    ImportedXsSummary result;

    for (const auto& vec : xs)
    {
        ASSERT_FALSE(vec.x.empty());
        ASSERT_EQ(vec.x.size(), vec.y.size());
        result.size.push_back(vec.x.size());
        result.x_bounds.push_back(vec.x.front());
        result.x_bounds.push_back(vec.x.back());
        result.y_bounds.push_back(vec.y.front() / units::barn);
        result.y_bounds.push_back(vec.y.back() / units::barn);
    }
    return result;
}

void ImportedXsSummary::print_expected()
{
    cout << "/*** ADD THE FOLLOWING UNIT TEST CODE ***/\n"
         << "static const real_type expected_size[] = "
         << repr(this->size) << ";\n"
         << "EXPECT_VEC_EQ(expected_size, result.size);\n"
         << "static const real_type expected_x_bounds[] = "
         << repr(this->x_bounds) << ";\n"
         << "EXPECT_VEC_SOFT_EQ(expected_x_bounds, result.x_bounds);\n"
         << "static const real_type expected_y_bounds[] = "
         << repr(this->y_bounds) << ";\n"
         << "EXPECT_VEC_SOFT_EQ(expected_y_bounds, result.y_bounds);\n"
         << "/*** END CODE ***/\n";
}
```

Note also that the `ASSERT_EQ` protects against a segfault if one of the vectors is empty.